nextcord/http.py

```diff
@@ -124,25 +140,329 @@ def bucket(self) -> str:
+
+    @property
+    def locked(self) -> bool:
+        return self.remaining <= 0
+
+    async def acquire(self) -> bool:
+        # If no more requests can be made but the event is set, clear it.
+        if self.locked and self._on_reset_event.is_set():
+            _log.debug(
+                "Bucket %s: Hit the remaining request limit of %s, locking until reset.",
+                self.bucket,
+                self.limit,
+            )
+            self._on_reset_event.clear()
+            if not self.resetting:
+                self.start_reset_task()
+
+        # Waits in a loop for the event to be set, clearing the event as needed and looping.
+        while not self._on_reset_event.is_set():
+            _log.debug("Bucket %s: Not set yet, waiting for it to be set.", self.bucket)
+            await self._on_reset_event.wait()
+
+            if self.locked and self._on_reset_event.is_set():
+                _log.debug(
+                    "Bucket %s: Hit the remaining limit of %s, locking until reset.",
+                    self.bucket,
+                    self.limit,
+                )
+                self._on_reset_event.clear()
+                if not self.resetting:
+                    self.start_reset_task()
+
+        if self.migrating:
+            raise RateLimitMigrating(
+                f"This RateLimit is deprecated, you need to migrate to bucket {self.migrating}"
+            )
+
+        _log.debug("Bucket %s: Continuing with request.", self.bucket)
+        self.remaining -= 1
+        return True
+
+    def release(self) -> None:
+        # Basically a placeholder, could probably be removed ;)
+        pass
+
+
+class GlobalRateLimit(RateLimit):
+    """
+    Represents the global rate limit, and thus has to have slightly modified behavior.
+
+    Still not thread safe.
+    """
+
+    def __init__(self, time_offset: float = 0.0, *args, **kwargs) -> None:
+        super().__init__(time_offset=time_offset, use_reset_timestamp=False)
+
+    async def acquire(self) -> bool:
+        ret = await super().acquire()
+        # As updates are little weird, it's best to start the reset task as soon as the first request has acquired.
+        if not self.resetting:
+            self.start_reset_task()
+
+        return ret
+
+    async def update(self, response: aiohttp.ClientResponse) -> None:
+        if (
+            response.headers.get("X-RateLimit-Global") != "true"
+            and response.headers.get("X-RateLimit-Scope") != "global"
+        ):
+            # The response is intended for the regular rate limit, not a global rate limit.
+            return
+
+        if response.status == 429:
+            # Oh dear, we hit the rate limit.
+            _log.warning("Global rate limit 429 encountered, setting remaining to 0.")
+            self.remaining = 0
+            if response.headers.get("X-RateLimit-Scope") == "global":
+                data = await response.json()
+                _log.debug(data)
```

Comment:
This is a sizeable slowdown for bots without logging enabled at the debug level. I assume this is a log leftover from development?

If it's not, please port it to a lazy logging setup
```suggestion
                _log.debug("%s", data)
```

Reply:
It was left over at development. I was going to remove it, but hitting the global rate limit is rare and debug logs in such a case would likely be useful.